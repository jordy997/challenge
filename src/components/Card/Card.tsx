import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

export type PointsProps = {
    points: number
}
const Card = ({ points }: PointsProps) => {
    return (
        <View style={styles.container}>
            <View style={styles.cardPoints}>
                <Text style={styles.month}>Month</Text>
                <Text style={styles.points}>{points} pts</Text>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        height: '25%'
    },
    cardPoints: {
        height: 143,
        width: 286,
        borderRadius: 20,
        marginTop: 7,
        alignSelf: 'center',
        backgroundColor: '#334FFA',
        position: 'absolute',
        elevation: 5,
        shadowOffset: {
            width: 0,
            height: 4,

        }
    },
    month: {
        fontWeight: '800',
        color: '#FFFFFF',
        fontSize: 16,
        lineHeight: 22,
        marginTop: 21,
        marginLeft: 18
    },
    points: {
        marginTop: 7,
        fontSize: 32,
        lineHeight: 44,
        fontWeight: '800',
        color: '#FFFFFF',
        textAlign: 'center'
    }
})

export default Card