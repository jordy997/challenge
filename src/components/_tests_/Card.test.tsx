import React from 'react'

import { shallow } from 'enzyme'
import Card from '../Card/Card'

let wrapper
describe('Card component', () => {

    beforeEach(() => {
        wrapper = shallow(<Card totalPoints={100000} />)
    })

    describe('Rendering', () => {
        it('Card should render correctly', () => {
            expect(wrapper).toMatchSnapshot()
        })
    })
})