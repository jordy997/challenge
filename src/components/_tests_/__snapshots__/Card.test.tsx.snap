// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Rendering Card should render correctly 1`] = `ShallowWrapper {}`;
